app/bundles/CoreBundle/Tests/Form/RequestTraitTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Tests\Form;
+
+use Mautic\CoreBundle\Form\RequestTrait;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Form;
+use Symfony\Component\Form\FormConfigBuilder;
+use Symfony\Component\Form\FormFactoryBuilder;
+
+class RequestTraitTest extends \PHPUnit\Framework\TestCase
+{
+    use RequestTrait;
+
+    /**
+     * @var Form
+     */
+    private $form;
+
+    protected function setUp(): void
+    {
+        $factoryBuilder = new FormFactoryBuilder();
+        $factoryBuilder->addType(new SubmitType());
+        $factoryBuilder->addType(new ButtonType());
+        $factory = $factoryBuilder->getFormFactory();
+
+        $dispatcher = $this->getMockBuilder(
+            'Symfony\Component\EventDispatcher\EventDispatcherInterface'
+        )->getMock();
+        $formConfigBuilder = new FormConfigBuilder('foo', null, $dispatcher);
+        $formConfigBuilder->setFormFactory($factory);
+        $formConfigBuilder->setCompound(true);
+        $formConfigBuilder->setDataMapper(
+            $this->getMockBuilder('Symfony\Component\Form\DataMapperInterface')->getMock()
+        );
+        $fooConfig = $formConfigBuilder->getFormConfig();
+
+        $this->form = new Form($fooConfig);
+    }
+
+    public function testMultiSelectPrepareParametersFromRequest()
+    {
+        $params = [
+            'multiselect'  => '',
+            'multiselect2' => 'input',
+            'multiselect3' => ['first'],
+            'multiselect4' => 'first|second',
+            'multiselect5' => ['first', 'second'],
+        ];
+
+        $expectedValues =
```

Comment:
This is just a suggestion for next PRs. Doesn't have to be implemented here. But PHPUNIT has Dataproviders for this use case.

https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers.examples.DataTest.php